src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -834,13 +834,15 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
 
         internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector, IEqualityComparer<K>? comparer = null)
             where K : notnull
+            where T : notnull
```

Comment:
it ensures we don't have to do any wonky `T => object?` or `object? => T` casts.  Since we only have two consumers and they both pass in non-null values, this was jsuta  nice easy path :)